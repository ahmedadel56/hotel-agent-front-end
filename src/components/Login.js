/* eslint-disable */
import React, { useState, useEffect } from 'react';
import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
import { logInUser } from '../redux/User/User';
import { useNavigate } from 'react-router-dom';
import auth from '../img/authentication.jpg';
import Alert from './Alert';

export default function Login() {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [success, setSucces] = useState(false);
  useEffect(() => {
    if (success) {
      const timeId = setTimeout(() => {
        setSucces(false);
      }, 3000);

      return () => {
        clearTimeout(timeId);
      };
    }
  }, [success]);

  const { register,formState: { errors }, handleSubmit } = useForm();
  const onSubmit =  (data) => {
    setSucces(true);
    const timeId = setTimeout(() => {
      dispatch(logInUser(data)) ? (navigate('/')) : null;
    }, 900);
    return () => {
      clearTimeout(timeId);
    };
  };

  return (
    <><>
      {success ? (
        <Alert type="success">
          <p>Welcome!</p>
        </Alert>
      ) : null}
    </><div
      className="reserve-container"
      style={{
        backgroundImage: `url(${auth})`,
      }}
    >
        <div className="reserve-heading">
          <h1>WELCOME BACK!</h1>
          <hr />
          <p>Sign in to continue</p>
        </div>
        <form onSubmit={handleSubmit(onSubmit)}>
          <div className="login-form">
            <input {...register('email', {
              required: "Email address is required",
              pattern: {
                value: /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
                message: "Invalid email address"
              }
            })} placeholder="✉️  email" />
            {errors.email && <p>{errors.email.message}</p>}
          </div>
          <div className="login-form">
            <input {...register('password')} type={'password'} required minLength={6} placeholder="🔒  password" />
          </div>
          <div className="login-btn">
            <input type="submit" value="SIGN IN" />
          </div>
        </form>
      </div></>
  );
}
/* eslint-enable */
