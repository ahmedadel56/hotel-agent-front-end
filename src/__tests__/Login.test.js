import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Login from '../components/Login';

it('Check if Login page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

afterEach(() => {
  jest.clearAllMocks();
});
