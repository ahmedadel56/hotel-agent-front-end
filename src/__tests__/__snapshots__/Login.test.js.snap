// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Login page renders correctly 1`] = `
<div
  className="reserve-container"
  style={
    Object {
      "backgroundImage": "url(authentication.jpg)",
    }
  }
>
  <div
    className="reserve-heading"
  >
    <h1>
      WELCOME BACK!
    </h1>
    <hr />
    <p>
      Sign in to continue
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="login-form"
    >
      <input
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="✉️  email"
      />
    </div>
    <div
      className="login-form"
    >
      <input
        minLength={6}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="🔒  password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="login-btn"
    >
      <input
        type="submit"
        value="SIGN IN"
      />
    </div>
  </form>
</div>
`;
